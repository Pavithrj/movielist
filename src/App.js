import React, { useEffect, useState } from 'react';
import Header from './Components/Header';
import Movie from './Components/Movie';
import './App.css';

const movieListUrl = "https://api.themoviedb.org/3/movie/top_rated?api_key=cc31d08b0d4b5b3539a406e5af2aec1f&language=en-US&page=1";
function App() {

  const [movies, setMovies] = useState([]);
  const [searchText, setSearchText] = useState("");
  const [filteredMovies, setFilteredMovies] = useState([]);

  const handleCountChange = (e) => {
    setSearchText(e.target.value);
  }

  const handleSearch = () => {
    const filteredMovies = movies.filter((movie) =>
      movie.title.toLowerCase().includes(searchText.toLowerCase())
    );
    setFilteredMovies([...filteredMovies]);
  };

  useEffect(() => {
    fetch(movieListUrl)
      .then((response) => response.json())
      .then((data) => {
        setMovies(data.results);
        setFilteredMovies(data.results);
      })
      .catch((error) => console.error(error));
  }, []);

  return (
    <div>
      <Header title="My Movie List" />
      <div>
        <input id='searchBox' type="text" value={searchText} onChange={handleCountChange} />
        <button id='searchButton' onClick={handleSearch}>Search</button>
      </div>
      <div className='movie-list'>
        {filteredMovies.map(item => <Movie movie={item} />)}
      </div>
    </div>
  );
}

export default App;