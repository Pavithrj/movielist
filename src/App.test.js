import React from 'react';
import { render, waitFor } from '@testing-library/react';
import { act } from "react-dom/test-utils";
import App from './App';

xdescribe('App Component', () => {
  beforeEach(() => {
    const mockData = {
      results: [
        {
          title: 'The GodFather',
          overview: 'abc',
          release_date: '1972-03-14'
        },
        {
          title: 'The Shawshank Redemption',
          overview: 'xyz',
          release_date: '1994-09-23'
        }
      ]
    };
    global.fetch = jest.fn().mockImplementation(() =>
      Promise.resolve({
        json: () => Promise.resolve(mockData)
      })
    );
  })

  it("should generate snapshot", async () => {
    let component;
    await act(async () => {
      component = await render(<App />);
    })
    expect(component.asFragment()).toMatchSnapshot();
  })

  it("should render movie name", async () => {
    const { getByText } = render(<App />);
    await waitFor(() => getByText('The GodFather'));
    expect(getByText('The GodFather')).toBeInTheDocument();
  })
})