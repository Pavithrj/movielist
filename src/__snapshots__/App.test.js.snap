// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should generate snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="headerStyle"
    >
      <h1>
        Movie Heist
      </h1>
    </div>
    <div
      class="movie-list"
    >
      <div
        class="movie box"
      >
        <img
          alt="movieposter"
          class="poster"
          src="https://image.tmdb.org/t/p/w342undefined"
        />
        <p
          class="title"
          title="The GodFather"
        >
          The GodFather
        </p>
        <p>
          Rating: 
        </p>
        <p>
          Release Date:1972-03-14
        </p>
        <div>
          <p
            class="description"
          >
            abc
          </p>
        </div>
        <div
          class="buttons"
        >
          <button
            class="cart"
          >
            ADD TO FAVORITE
          </button>
        </div>
      </div>
      <div
        class="movie box"
      >
        <img
          alt="movieposter"
          class="poster"
          src="https://image.tmdb.org/t/p/w342undefined"
        />
        <p
          class="title"
          title="The Shawshank Redemption"
        >
          The Shawshank Redemption
        </p>
        <p>
          Rating: 
        </p>
        <p>
          Release Date:1994-09-23
        </p>
        <div>
          <p
            class="description"
          >
            xyz
          </p>
        </div>
        <div
          class="buttons"
        >
          <button
            class="cart"
          >
            ADD TO FAVORITE
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
